samples/event-sourcing-esdb-simple/src/main/java/io/eventdriven/ecommerce/core/projections/JPAProjection.java

```diff
@@ -0,0 +1,60 @@
+package io.eventdriven.ecommerce.core.projections;
+
+import io.eventdriven.ecommerce.core.events.EventEnvelope;
+import io.eventdriven.ecommerce.core.views.VersionedView;
+import org.springframework.data.repository.CrudRepository;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public abstract class JPAProjection<View, Id> {
+  private final CrudRepository<View, Id> repository;
+
+  protected JPAProjection(CrudRepository<View, Id> repository) {
+    this.repository = repository;
+  }
+
+  protected <Event> void add(EventEnvelope<Event> eventEnvelope, Supplier<View> handle) {
+    var result = handle.get();
+
+    if(result instanceof VersionedView versionedView){
+      versionedView.setMetadata(eventEnvelope.metadata());
+    }
+
+    repository.save(result);
+  }
+
+  protected <Event> void GetAndUpdate(
```

Comment:
Method starts with upper case... But I know that you know :D I have exactly the same when writing in C#

Reply:
Good catch, fixed in https://github.com/oskardudycz/EventSourcing.JVM/pull/1/commits/f064e1892b59d153e7294846e07720a756f1447b 👍